### ISSUE TITLE
Provide a means to check if the API is supported by the platform or OS

### ISSUE THREAD

**Issue by marcoscaceres** [w3c/web-share#286]
The current spec presumes that if the OS or user agent doesn't support a sharing capability or intent, then the API shouldn't be exposed... the problem is that this can then be misused with other features to potentially detect a user agent type or platform.

What we should actually have is a dedicated way to check if sharing is supported.

**Comment by saschanaz** [comment]
> The OS/browser/user should be allowed to lie should it choose to for whatever reason (e.g., the user is allergic to share sheets).

This seems to be the real intent than the platform support indication as described by the title, is that correct? (Otherwise I'm not sure why the current behavior is insufficient, as there are easier ways to detect platforms, namely `navigator.userAgent`.)

Assuming that, I'm not sure such user intent should be exposed too easily, because that would be a good fingerprinting vector.

**Comment by marcoscaceres** [comment]
Yes, exactly. It should guard against exposing any user preference (or be obscure enough that it can't be used to fingerprint). 

You are correct that `navigator.userAgent` provides some this, but, for instance, it's shouldn't be possible to detect if you are on an iPhone or an iPad via `navigator.userAgent`.

**Comment by jcayzac** [comment]
@marcoscaceres One issue I have with the API in its current form, as a developer, is that there is no way to query if the share feature is _usable_ (not _available_) on the platform.

For instance, here is a share sheet that offers basically zero benefit to the end user, and that I (as a developer) would likely not want to use (and switch to using the clipboard API instead)[^1]:

<img width="254" height="146" alt="Image" src="https://github.com/user-attachments/assets/c19034ca-2b64-459b-bae0-6b2d29d51eb5" />

Some basic usability querying such as:

1. Does the share UI offer at least a "copy to clipboard" fallback choice?
2. Are there any sharing outlets (i.e. apps) actually registered on the system that the share API is useful?

(since 2 could be used as a fingerprinting signal, the user agent could maybe return an order of magnitude for the number of apps, e.g. 1 for a few, 10 for a lot).

Slightly off-topic, but since this issue is about capability querying I thought it was worth adding here. Some information could even be returned by the current `canShare()` while being backward compatible with the current semantics, since any object is a truish value.

[^1]: Right now, the only way to do something close to that is to use `matchMedia` to try to detect mobile devices and enable "share" only there, and "copy" elsewhere[^2]. Not great.
[^2]: Both "copy" and "share" are the same feature, really. The user wants to move data somewhere. I feel the web APIs got caught into 1980s desktop & 2000s social networks UX concepts instead of offering forward-looking generalizations.

**Comment by saschanaz** [comment]
> 1. Does the share UI offer at least a "copy to clipboard" fallback choice?

There's no real great way to tell this from browser engine side either, other than having some heuristic based on OS version. Because the platform APIs don't tell that either.

IMO the browser should just provide it regardless of the underlying platform implementation, as Edge does.

> 2. Are there any sharing outlets (i.e. apps) actually registered on the system that the share API is useful?
>
> (since 2 could be used as a fingerprinting signal, the user agent could maybe return an order of magnitude for the number of apps, e.g. 1 for a few, 10 for a lot).

This does sound scary... and the number of apps don't really mean any of those would be "useful", which depends a lot on what is being shared and what those apps do with it.
